test/testdata/packager/export_all/__package.rb

```diff
@@ -0,0 +1,11 @@
+# frozen_string_literal: true
+# typed: strict
+# enable-packager: true
+
+class Foo::Bar < PackageSpec
+  export_all!
+
+  export Foo::Bar::Thing
+       # ^^^^^^^^^^^^^^^ error: Package `Foo::Bar` declares `export_all!` and therefore should not use explicit exports
```

Comment:
OOC, does this same error happen if the ordering of these declarations is reversed?

Reply:
Yes: the parsing logic accumulates exports first, and then these errors are raised only after the entire `__package.rb` file has been looked at.